core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -74,6 +75,16 @@ object Json {
     catch { case e: JsonProcessingException => Left(e) }
   }
 
+  /**
+    * Parse a JSON string into a JsonValue if possible. It returns a `Try` in case any exception happens.
+    * @param input a JSON string to parse
+    * @return A `Try` which in case of `Success` contains either `None` if the parsing was unsuccessful or `Some(x)` if
+    *         the parsing was successful where `x` is a `JsonValue`.
+    */
+  def tryParseFull(input: String): Try[Option[JsonValue]] = Try(doParseFull(input))
```

Comment:
like other methods can we return Either[JsonProcessingException, Option[JsonValue]] ? it will be clean to handle JsonProcessingException alone.  maybe ismael can advice here.

Reply:
Refactored it, I agree that we should keep it consistent.